walnut/converter/converter.go

```diff
@@ -139,19 +144,50 @@ func TaskToYAML(extractedTasks []*task.Class) (err error) {
 	}
 
 	for _, SingleTask := range extractedTasks {
-		YAMLData, err := yaml.Marshal(SingleTask)
+		YAMLData, err := yaml.Marshal(&SingleTask)
 		if err != nil {
 			return fmt.Errorf("marshal failed: %v", err)
 		}
 		fileName := "tasks/" + SingleTask.Identifier.Name + ".yaml"
 		f, err := os.Create(fileName)
-		defer f.Close()
```

Comment:
This was actually better than moving the `f.Close()` to the bottom, because now if you hit the `err` case you're leaving an open file handle.

Reply:
Fair point. I thought having an `defer` inside a for loop is [a bad idea](https://blog.golang.org/defer-panic-and-recover), but the orphaned file handle is probably worse. I'll have it fixed.